server/controllers/orderController.ts

```diff
@@ -68,19 +70,18 @@ export default class OrderController {
   submit: RequestHandler = async (req: Request, res: Response) => {
     const order = req.order!
 
-    if (order.status === 'SUBMITTED') {
-      res.redirect(paths.ORDER.SUBMIT_FAILED.replace(':orderId', order.id))
-    } else {
-      try {
-        const { token } = res.locals.user
-        await this.orderService.submitOrder({ accessToken: token, orderId: order.id })
-
-        res.redirect(paths.ORDER.SUBMIT_SUCCESS.replace(':orderId', order.id))
-      } catch (error) {
-        req.flash('validationErrors', error)
+    const result = await this.orderService.submitOrder({
+      orderId: order.id,
+      accessToken: res.locals.user.token,
+    })
 
-        res.redirect(paths.ORDER.SUBMIT_FAILED)
-      }
+    if (result.submitted) {
+      res.redirect(paths.ORDER.SUBMIT_SUCCESS.replace(':orderId', order.id))
+    } else if (result.type === 'alreadySubmitted' || result.type === 'incomplete') {
+      req.flash('submissionError', result.error)
+      res.redirect(paths.ORDER.SUMMARY.replace(':orderId', order.id))
+    } else if (result.type === 'errorStatus') {
+      res.redirect(paths.ORDER.SUBMIT_FAILED.replace(':orderId', order.id))
     }
```

Comment:
I think you're missing an else condition

Reply:
Thanks Ben. As the fallback `else` will be used when an unaccounted-for condition occurs, I've set it to route the user to the summary page. Does this seem suitable?
An alternative would be to route to the summary page & display an 'unknown error' message.

Reply:
Yeap, maybe add something like `req.flash('submissionError', "Oops, something went wrong")` to that else so that the user has some feedback, and then run it past Rachel

Reply:
I'm pretty sure that else is unreachable anyway, but just on the off chance it isn't, we need to make sure we're sending a response.

Reply:
Thanks Ben. I raised this in today's build review. We talked about it but didn't settle on a solution.

I've implemented a temporary fallback, which routes to the summary screen with the error message "Something unexpected happened. Please try again in a few minutes." This is a placeholder & probably isn't the final implementation.

I'll raise this question with Emma, who is considering content for unhappy paths, and I'll pop a ticket on the board to make sure this isn't forgotten.